drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1605,6 +1624,8 @@ static void eth0_iface_init(struct net_if *iface)
 #ifdef CONFIG_ETH_SAM_GMAC_MAC_I2C_EEPROM
 	/* Read MAC address from an external EEPROM */
 	get_mac_addr_from_i2c_eeprom(dev_data->mac_addr);
+#elif defined(CONFIG_ETH_SAM_GMAC_RANDOM_MAC)
```

Comment:
No objection to this feature but could we do it like this instead:
```
generate_mac(addr)
{
#ifdef CONFIG_ETH_SAM_GMAC_MAC_I2C_EEPROM
    get_mac_addr_from_i2c_eeprom(dev_data->mac_addr);
#elif defined(CONFIG_ETH_SAM_GMAC_RANDOM_MAC)
here would be your code
#endif 
}
```

and then we would just call `generate_mac()` from `eth0_iface_init()`